ludwig/utils/misc_utils.py

```diff
@@ -100,7 +100,14 @@ def set_default_value(dictionary, key, value):
 def set_default_values(dictionary, default_value_dictionary):
```

Comment:
nit: Either in this PR (or add a TODO) to clarify what `set_default_values` does now, with a better docstring and additional tests that demonstrates it handling recursive/empty dictionaries.